File: add-strings/add-strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) 
    {
        int carry=0;
        string result=""; //create new empty string
        int i= num1.size()-1;//point to last index of each string
        int j= num2.size()-1;
        while(i>=0 || j>=0)
        {
            int sum=0; 
            if(i>=0)
            {
                sum += (num1[i]-'0'); //convert string element to ASCII value and see if its in front of ASCII value of '0'
                i--; //go to next element
            }
            if(j>=0)
            {
                sum += (num2[j]-'0');//convert string element to ASCII value and see if its in front of ASCII value of '0'
                j--; //go to next element
            }
            sum+=carry; //sum will always have the value of carry 
            carry=sum/10; //to get the number carried to next element (Quotient)
            sum=sum%10; //remainder left gets added to the new string as normal addition
            result=result+to_string(sum); //adding the remainder here
        }
       if(carry!=0) //Edge case where a carry exists if both strings are of the same length 
        {
           result=result+to_string(carry); //if theres a carry still left then add it to the back of the string
        }
        reverse(result.begin(), result.end()); //since we add all elements to back of the string we need to reverse it to get the final answer
        return result;
    }
};

File: add-strings/add-strings.java
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuilder result=new StringBuilder(); //make a new result string
        int carry=0;
        int i=num1.length()-1; //start from last index of both strings
        int j=num2.length()-1;
        while(i>=0 || j>=0) //going from last index to the first
        {
            int sum=carry; //in the begining sum will be 0 since no addition has taken place and no carry
            //if(i>=0 and j>=0) run for all elements of the loop and assuming one string has more elemts left it then adds those numbers
            
            if(i>=0) //remaining elements present in string 1
            {
                sum+=num1.charAt(i)-'0'; //adding the numbers by converting the to charecter and subtracting '0' from the ASCII equivalent, thus giving us an integer
                i--; //to go to the next iteration/element
            }
            if(j>=0)//remaining elements present in string 2
            {
                sum+=num2.charAt(j)-'0'; //adding the numbers by converting the to charecter and subtracting '0' from the ASCII equivalent, thus giving us an integer
                j--;//to go to the next iteration/element
            }
            result.append(sum%10); //adding the remainder to the end of the result
            carry=sum/10; //to find the value to be transfered to next position
            
        }
        if(carry!=0) //if after doing the addition a carry still remains we add it to the end
             result.append(carry);
        return result.reverse().toString(); //the result string we would get is 431 as in example1 so to get the right value we reverse it
    }
}